src/app/cluster/details/cluster/component.ts

```diff
@@ -527,6 +530,25 @@ export class ClusterDetailsComponent implements OnInit, OnDestroy {
     return !!this.cluster?.spec?.cniPlugin && this.cluster?.spec?.cniPlugin?.type !== CNIPlugin.None;
   }
 
+  toggleTerminal() {
+    if (!this.isClusterRunning) {
+      return;
+    }
+    this.showTerminal = !this.showTerminal;
+  }
+
+  onCloseTerminal(isClosed: boolean) {
+    if (isClosed) {
```

Comment:
Added for type safety reason. But it will always be false and secondly we are sure that user has pressed the close button either from toolbar/status bar so it will be `true` means -> user has performed action to close terminal which is `true` 
I have got your point it is fixed: [`2476044` (#4492)](https://github.com/kubermatic/dashboard/pull/4492/commits/2476044226467a03f740c2a7a0800d1a93b71c76)